### PR TITLE
feat: preferences page, styling, and routing between onboarding

### PR DESCRIPTION
**Pull request by me-liu**
[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- changed the global styling to remove default margins/whitespace. This may mean previously styled pages must be changed
- preferences page styling
- navigation between the preferences and onboarding page

### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Screenshot 2024-10-12 at 4 46 36 PM

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- All at once: the styling is almost exactly the same as my previous PR

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- I will add more global colors soon

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii

### CODE REVIEW

File: app/preferences/styles.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import styled from 'styled-components';
+import COLORS from '../../styles/colors';
+
+export const Background = styled.main`
+  flex-direction: column;
+  min-width: 100%;
+  min-height: 100svh;
+  display: flex;
+  align-items: center;
+  justify-content: start;
```
Comment (celinechoiii) [inline app/preferences/styles.ts]: `justify-content: center`

File: app/preferences/styles.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+export const InputText = styled.text`
+  margin-top: 6%;
+`;
+
+export const Input = styled.input`
+  width: 100%;
+  height: 22px;
+  border-color: ${COLORS.gray4};
+  border-style: solid;
+  border-radius: 4px;
```
Comment (celinechoiii) [inline app/preferences/styles.ts]:
```
margin-bottom: 1.3rem;
margin-top: 0.25rem;
padding-left: 0.4rem;
```
for font-size, padding, and margin, it is good practice to use `rem` or `em` since they help create responsive designs! `rem` is relative to the root (`html`) element's font size and `em` is relative to the parent element's font size. i added `margin-top` to ensure that we have more space between the text and the input box. `padding-left` helps add space between the cursor and the left side of the container in your `Input` component. i recommend you read over the styling guide in our b&r notion page, and if you have any questions, please let me know!

**Comment by celinechoiii** [comment]
since all of the styled components from `preferences` screen overlap with the `general` screen, can you have a single `styles.ts` file that can be used across all of the onboarding flow to reduce redundancy? also, could you rearrange the files like the following: `onboarding` folder --> `general` folder (contains page.tsx) + `preferences` folder (contains page.tsx) + `styles.ts`. if there are any screen specific styled components you would like to add, feel free to add a `styled.ts` file for that respective folder

File: app/preferences/styles.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  flex-direction: column;
+  align-items: start;
+  margin: 25px 0px;
+  justify-content: space-between;
+  background-color: white;
+  border-radius: 8px;
+  padding: 13%;
+`;
+
+export const InputText = styled.text`
```
Comment (celinechoiii) [inline app/preferences/styles.ts]: remove the `InputText` component and use `<text>` tag

File: app/preferences/page.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          <Image src={Back} alt="Back icon" />
+        </Link>
+
+        <div>
+          <Rectangle variant="light" widthPercentage="50%" />
+          <Rectangle variant="dark" widthPercentage="50%" />
+        </div>
+        <Container>
+          <Title>Help us tailor shows to you!</Title>
+          <InputText> Facility Type</InputText>
```
Comment (celinechoiii) [inline app/preferences/page.tsx]: remove the white spaces before `InputText` for all of the preferences fields

File: app/preferences/styles.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  border-color: ${COLORS.gray4};
+  border-style: solid;
+  border-radius: 4px;
+`;
+
+export const Title = styled.h1`
+  font-size: 28px;
+  text-align: start;
+  color: ${COLORS.gray11};
+  margin-top: 0;
```
Comment (celinechoiii) [inline app/preferences/styles.ts]: delete `margin-top` and add `margin-bottom: 2.25rem;` to add more space between title and the first input field

File: app/preferences/styles.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  padding: 13%;
+`;
+
+export const InputText = styled.text`
+  margin-top: 6%;
+`;
+
+export const Input = styled.input`
+  width: 100%;
+  height: 22px;
```
Comment (celinechoiii) [inline app/preferences/styles.ts]: adjust height to `25px`

File: app/onboarding/page.tsx [hunk 7]
```diff
@@ -52,11 +53,14 @@ export default function Onboarding() {
             </UpdateText>
           </UpdateContainer>
         </Container>
-        <ButtonContainer>
-          <ContinueButton>
```
Comment (celinechoiii) [inline app/onboarding/page.tsx:56]: for some reason, when i click on the continue button, there is a red underline? can you try fixing this

**Comment by celinechoiii** [comment]
can you make sure to pull from main? i merged in justin's font pr and now the styling might need some adjustments

File: app/preferences/styles.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  width: ${({ widthPercentage }) => widthPercentage};
+  height: 7px;
+  display: inline-block;
+  background: ${({ variant }) =>
+    variant === 'light' ? COLORS.gray8 : COLORS.gray10};
+  border-radius: 2px;
+`;
+
+export const Container = styled.main`
+  width: 100%;
```
Comment (celinechoiii) [inline app/preferences/styles.ts]: delete `width` and `height`

File: app/preferences/styles.ts (hunk 5 above)
Comment (me-liu) [reply app/preferences/styles.ts]: The reason I did margin-top: 0 because the title text has some default margins on the top and bottom. I think if do this I end up with  a lot of spacing btween the title and the rest of the stuff: 
Screenshot 2024-10-19 at 5 35 28 PM

File: app/onboarding/general/page.tsx [hunk 9]
```diff
@@ -52,11 +49,16 @@ export default function Onboarding() {
...
-        <ButtonContainer>
-          <ContinueButton>
-            <ContinueText>Continue</ContinueText>
-          </ContinueButton>
-        </ButtonContainer>
+
+        <StyledLink href="../onboarding/preferences">
+          <ButtonContainer>
+            <ContinueButton>
+              <ContinueText style={{ fontFamily: 'boo' }}>
```
Comment (celinechoiii) [inline app/onboarding/general/page.tsx]: delete the inline styling

File: app/onboarding/styles.ts [hunk 10]
```diff
@@ -111,14 +96,18 @@ export const ContinueButton = styled.button`
...
+  text-decoration: none;
 `;
 
-export const ContinueText = styled.p`
+export const ContinueText = styled.text`
   color: white;
   font-size: 14px;
   padding: 10px;
+  text-decoration: none;
+  font-family: 'BespokeSans-Regular' !important;
```
Comment (celinechoiii) [inline app/onboarding/styles.ts]: delete `font-family`

File: app/onboarding/styles.ts [hunk 11]
```diff
@@ -111,14 +96,18 @@ export const ContinueButton = styled.button`
   justify-content: center;
   align-items: center;
   gap: 10px;
+  cursor: pointer;
+  text-decoration: none;
 `;
 
-export const ContinueText = styled.p`
+export const ContinueText = styled.text`
```
Comment (celinechoiii) [inline app/onboarding/styles.ts:103]: `import { BespokeSans } from '@/styles/fonts';` at the top of the document

under the `ContinueText` styling, add `${BespokeSans.style}`

File: app/onboarding/styles.ts [hunk 12]
```diff
@@ -19,6 +19,8 @@ export const InlineContainer = styled.main`
   width: 30%;
   height: 85%;
```
Comment (celinechoiii) [inline app/onboarding/styles.ts]: delete `height`
